static/js/lib/fetch.js

```diff
@@ -0,0 +1,381 @@
+(function() {
```

Comment:
Where's this dep coming from and does it have a version? Couldn't this come from a npm package so we can manage the versions + updates?
What's the rationale for fetch versus using jQuery which is already being used?

Not sure what our present front-end browser support strategy is for AMO but IE has no promise support at all [1] so would this also need a promise shim for that and/or older versions?

[1] http://caniuse.com/#search=Promise